gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiStateServiceImpl.java

```diff
@@ -284,6 +315,34 @@ public GenericApiEntity stop(ExecutionContext executionContext, String apiId, St
         }
     }
 
+    @Override
+    public boolean stopV2DynamicProperties(String apiId, EventManager eventManager) {
+        try {
+            log.debug("Stopping V2 Dynamic properties for API {}", apiId);
+            Api apiToUpdate = apiRepository.findById(apiId).orElseThrow(() -> new ApiNotFoundException(apiId));
+
+            eventManager.publishEvent(ApiEvent.STOP_DYNAMIC_PROPERTY_V2, apiToUpdate);
+            return true;
+        } catch (TechnicalException ex) {
+            log.error("An error occurs while trying to stop V2 Dynamic properties for API {}", apiId, ex);
+            throw new TechnicalManagementException("An error occurs while trying to stop V2 Dynamic properties for API " + apiId, ex);
+        }
+    }
+
+    @Override
+    public boolean stopV4DynamicProperties(String apiId, EventManager eventManager) {
+        try {
+            log.debug("stopping V4 Dynamic properties for API {}", apiId);
+            Api apiToUpdate = apiRepository.findById(apiId).orElseThrow(() -> new ApiNotFoundException(apiId));
+
+            eventManager.publishEvent(ApiEvent.STOP_DYNAMIC_PROPERTY_V4, apiToUpdate);
+            return true;
+        } catch (TechnicalException ex) {
+            log.error("An error occurs while trying to stop V4 Dynamic properties for API {}", apiId, ex);
```

Comment:
do not log if we rethrow an exception https://gravitee.slab.com/posts/good-practices-for-handling-java-exceptions-and-logging-1dc30qlp